### PR TITLE
Refactor packages

### PR DESCRIPTION
**Pull request by nutzhub**
Aim to simplify and clear unused code
- Focus only s3 emitter
- Reduce 3rd deps

### CODE REVIEW

File: buffer_test.go [hunk 1]
```diff
@@ -28,10 +27,15 @@ func Test_FirstSeq(t *testing.T) {
 	r2 := &kinesis.Record{SequenceNumber: &s2}
 
 	b.AddRecord(r1)
-	assert.Equal(t, b.FirstSeq(), "1")
+	if b.FirstSeq() != "1" {
+		t.Fail()
+	}
+	// assert.Equal(t, b.FirstSeq(), "1")
```
Comment (aunsira) [inline buffer_test.go]: commented code

File: consumer.go [hunk 2]
```diff
@@ -90,7 +71,9 @@ func (c *Consumer) handlerLoop(shardID string, handler Handler) {
 		)
 
 		if err != nil {
-			log.Fatalf("Error GetRecords %v", err)
+			ctx.WithError(err).Error("GetRecords")
+			shardIterator = c.getShardIterator(shardID)
+			continue
```
Comment (aunsira) [inline consumer.go]: Why there's `continue` here?

File: consumer.go [hunk 3]
```diff
@@ -90,7 +71,9 @@ func (c *Consumer) handlerLoop(shardID string, handler Handler) {
 		)
 
 		if err != nil {
-			log.Fatalf("Error GetRecords %v", err)
+			ctx.WithError(err).Error("GetRecords")
+			shardIterator = c.getShardIterator(shardID)
```
Comment (aunsira) [inline consumer.go]: Duplicated with line#62

**Review by aunsira — COMMENTED** [review]
One more thing about repository license. Aon has raised this issue because it's possible that auditors will scan through our code and this derived from public repo. (haven't much details about this,  please ask Aon for more.) 

Edit. Original repo using `MIT License`. Might be alright to use.

File: consumer.go (hunk 2 above)
Comment (nutzhub) [reply consumer.go]: Hmm that is mess code from author. I will add more commit

**Review by aunsira — APPROVED** [review]

**Pull request merged**